### PR TITLE
Add SLSA L3 provenance to the sigbuild Dockers

### LINKED ISSUE #125: Enhancement: Add signed SLSA attestations to nightly docker builds
Hi! I'm reaching out on behalf of the [Open Source Security Foundation (openssf.org)](https://openssf.org/). We work on improving supply-chain security of critical open source projects. We believe we can help improve the tamper resistance of TF's nightly docker builds, with just a few lines of code. Starting with nightly docker builds is relatively low-risk, and allows some testing of the verification flow before moving to the more stable images.

[These](https://github.com/tensorflow/build/blob/428ab6533bdbcc9f8c1f2598825d8a2367368810/tf_sig_build_dockerfiles/README.md#tags) docker builds are built in this repository via GitHub Actions. Adding a  [provenance attestation](https://slsa.dev/provenance/v0.2) during the build process allows a cryptographically verifiable guarantee that the image was built in this repository. This provenance is a file with [metadata](https://github.com/slsa-framework/slsa-github-generator/tree/main/internal/builders/container#provenance-example)  that lets users know that the container images were built from your repository’s [workflow](https://github.com/tensorflow/build/blob/491856aa90413f54f6d442a2f04bc9ea8c96f337/.github/workflows/docker.yml) and not altered by anyone. 

The container provenance generator [workflow](https://github.com/slsa-framework/slsa-github-generator/tree/main/internal/builders/container) uploads the attestation in the registry using Sigstore's attestation [specification](https://github.com/sigstore/cosign/blob/92ce88e0fa077b50de9f9fe8772cdc893d58ca38/specs/ATTESTATION_SPEC.md). This is projected to go GA in the next few weeks (the API may be change in future releases of the generator), and we would love to use TF's nightly builds as an early adoption.

Generating these attestations is also a step toward adopting Supply-chain Levels for Software Artifacts ([SLSA](slsa.dev)). SLSA is a security framework to improve transparency and authenticity of the build / release process. It’s designed and used by a consortium of companies including Google, Intel, Chainguard, Citi, and Datadog, under the umbrella of the Open Source Security Foundation. By generating these attestations in the future for stable docker builds, your project will reach [SLSA Level 3](https://slsa.dev/spec/v0.1/levels#summary-of-levels) for provenance. You can even add a SLSA badge to your repository so users know that you take security seriously. 

If you're interested in reading more, check out this recent [blog post](https://slsa.dev/blog/2022/06/slsa-github-workflows), which describes how some of the recent supply-chain attacks would have been prevented using SLSA provenance.

Happy to answer any question you might have.

I'll follow up with a PR and link here to show the changes needed to add provenance to the docker workflow.

cc @ianlewis @mihaimaruseac @laurentsimon

### PR DESCRIPTION
**Pull request by mihaimaruseac**
Should close https://github.com/tensorflow/build/issues/125

TODO: the same for all other containers we maintain, maybe?

Signed-off-by: Mihai Maruseac <mihaimaruseac@google.com>

### CODE REVIEW

**Comment by angerson** [comment]
Cool, thanks!

**Review by angerson — APPROVED** [review]

File: .github/workflows/sigbuild-docker-branch.yml
Comment (angerson) [inline .github/workflows/sigbuild-docker-branch.yml:91]: Is this `build` rather than the step id, `docker_build`?

**Comment by mihaimaruseac** [comment]
Oh, still need to work on this, sorry. I'll remove the draft state when ready.

**Comment by bhack** [comment]
Can you verify @pranavladkat's claim at https://github.com/tensorflow/tensorflow/issues/58310#issuecomment-1293031713?

File: .github/workflows/sigbuild-docker-branch.yml
Comment (mihaimaruseac) [reply .github/workflows/sigbuild-docker-branch.yml:91]: Yep, fixed

**Comment by mihaimaruseac** [comment]
So, this is blocked because our docker username is encoded in secrets instead of directly in the workflow. Will need to fix this first.

**Comment by ianlewis** [comment]
> So, this is blocked because our docker username is encoded in secrets instead of directly in the workflow. Will need to fix this first.

https://github.com/slsa-framework/slsa-github-generator/pull/1409 should unblock this once it's merged and released.

**Comment by gbaned** [comment]
@mihaimaruseac This PR is in draft, any update on this? Please. Thank you!

**Comment by ianlewis** [comment]
> @mihaimaruseac This PR is in draft, any update on this? Please. Thank you!

The [repository-username](https://github.com/slsa-framework/slsa-github-generator/blob/v1.5.0/internal/builders/container/README.md#workflow-inputs) flag was released in v1.5.0 so you should be able to upgrade and use that secret.
https://github.com/slsa-framework/slsa-github-generator/blob/v1.5.0/CHANGELOG.md#v150

**Comment by mihaimaruseac** [comment]
Thanks. I'll find some time and update soon

**Comment by gbaned** [comment]
> @mihaimaruseac This PR is in draft, any update on this? Please. Thank you!

> Thanks. I'll find some time and update soon

Hi @mihaimaruseac Any update on this PR? Please. Thank you!

**Comment by mihaimaruseac** [comment]
The containers have changed and also I heard rumors that TF no longer cares about security, so this has been downprioritized

**Comment by mihaimaruseac** [comment]
Closing this one since the codebase changed a lot and will revisit when there is time.

**Comment by mihaimaruseac** [comment]
(please don't delete this branch)

**Comment by bhack** [comment]
I've just restored it. :+1:

**Pull request closed without merging**
